File: app/lib/models/venue.dart
class Venue {
  final String id;
  final String name;
  final String address;
  final String city;
  final String country;
  final double lat;
  final double lng;
  final String? surface;
  final String? pitchSize;
  final bool? parking;
  final String? photoUrl;

  const Venue({
    required this.id,
    required this.name,
    required this.address,
    required this.city,
    this.country = 'Ireland',
    required this.lat,
    required this.lng,
    this.surface,
    this.pitchSize,
    this.parking,
    this.photoUrl,
  });

  factory Venue.fromJson(Map<String, dynamic> json) {
    return Venue(
      id: json['id'] as String,
      name: json['name'] as String,
      address: json['address'] as String,
      city: json['city'] as String,
      country: json['country'] as String? ?? 'Ireland',
      lat: (json['lat'] as num).toDouble(),
      lng: (json['lng'] as num).toDouble(),
      surface: json['surface'] as String?,
      pitchSize: json['pitch_size'] as String?,
      parking: json['parking'] as bool?,
      photoUrl: json['photo_url'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'address': address,
        'city': city,
        'country': country,
        'lat': lat,
        'lng': lng,
        'surface': surface,
        'pitch_size': pitchSize,
        'parking': parking,
        'photo_url': photoUrl,
      };
}

File: api/models/venue.py
import uuid

from sqlalchemy import Float, String, func
from sqlalchemy.dialects.postgresql import UUID as PG_UUID
from sqlalchemy.orm import Mapped, mapped_column

from database import Base


class Venue(Base):
    __tablename__ = "venues"

    id: Mapped[uuid.UUID] = mapped_column(
        PG_UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    address: Mapped[str] = mapped_column(String(500), nullable=False)
    city: Mapped[str] = mapped_column(String(100), nullable=False)
    country: Mapped[str] = mapped_column(String(100), default="Ireland")
    lat: Mapped[float] = mapped_column(Float, nullable=False)
    lng: Mapped[float] = mapped_column(Float, nullable=False)
    surface: Mapped[str | None] = mapped_column(String(50))  # astro, grass, indoor
    pitch_size: Mapped[str | None] = mapped_column(String(20))  # 5-a-side, 7-a-side
    parking: Mapped[bool | None] = mapped_column(default=None)
    photo_url: Mapped[str | None] = mapped_column(String(500))
    created_by: Mapped[uuid.UUID | None] = mapped_column(PG_UUID(as_uuid=True))
    created_at: Mapped[str] = mapped_column(server_default=func.now())
